app/src/main/java/com/infomaniak/drive/ui/MainActivity.kt

```diff
@@ -366,29 +367,38 @@ class MainActivity : BaseActivity() {
         if (UploadFile.getAppSyncSettings()?.deleteAfterSync == true && UploadFile.getCurrentUserPendingUploadsCount() == 0) {
             UploadFile.getAllUploadedFiles()?.let { filesUploadedRecently ->
                 if (filesUploadedRecently.size >= SYNCED_FILES_DELETION_FILES_AMOUNT) {
-                    Utils.createConfirmation(
-                        context = this,
-                        title = getString(R.string.modalDeletePhotosTitle),
-                        message = getString(R.string.modalDeletePhotosNumericDescription, filesUploadedRecently.size),
-                        buttonText = getString(R.string.buttonDelete),
-                        isDeletion = true,
-                    ) {
+
+                    fun getFilesUriToDelete(uploadFiles: List<UploadFile>): List<Uri> {
+                        return uploadFiles
+                            .filter {
+                                !it.getUriObject().scheme.equals(ContentResolver.SCHEME_FILE) &&
+                                        !DocumentsContract.isDocumentUri(this, it.getUriObject())
+                            }
+                            .map { it.getUriObject() }
+                    }
+
+                    fun onConfirmation(filesUploadedRecently: ArrayList<UploadFile>, filesUriToDelete: List<Uri>) {
                         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
-                            val filesDeletionRequest = MediaStore.createDeleteRequest(
-                                contentResolver,
-                                filesUploadedRecently
-                                    .filter {
-                                        !it.getUriObject().scheme.equals(ContentResolver.SCHEME_FILE) &&
-                                                !DocumentsContract.isDocumentUri(this, it.getUriObject())
-                                    }
-                                    .map { it.getUriObject() }
-                            )
+                            val filesDeletionRequest = MediaStore.createDeleteRequest(contentResolver, filesUriToDelete)
                             uploadedFilesToDelete = filesUploadedRecently
                             filesDeletionResult.launch(IntentSenderRequest.Builder(filesDeletionRequest.intentSender).build())
                         } else {
                             mainViewModel.deleteSynchronizedFilesOnDevice(filesUploadedRecently)
                         }
                     }
+
+                    // We check that the filtered list of URIs is not empty before showing the dialog and sending the request to
+                    // MediaStore; otherwise, it would cause a crash.
```

Comment:
I approve this PR, but I feel there is too much indentation :

<img width="1054" alt="image" src="https://github.com/user-attachments/assets/4452304d-2e0d-4734-a753-b5d8e3527ac8">

Reply:
I've pushed another PR to enhance that : #1485.

Reply:
Result :

<img width="1046" alt="image" src="https://github.com/user-attachments/assets/fffecde0-5576-4c8b-8157-3cab63c2cfdb">

Reply:
Result :

<img width="1057" alt="image" src="https://github.com/user-attachments/assets/25074795-44ff-4a0f-9eac-37f3ba871ce0">

Reply:
Result :

<img width="1072" alt="image" src="https://github.com/user-attachments/assets/a81bb163-9ecc-4aaf-9661-45b59c4c53cf">